.gitignore

```diff
@@ -1,2 +1,17 @@
+# Generated by Cargo
+# will have compiled files and executables
+**/target/
 /target
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+# Unneeded artifacts
+**/.DS_Store
+
+# IDE artifacts
+/.idea
+/.vscode
+
+# Why we remove Cargo.lock?
```

Comment:
I also think we should keep the `Cargo.lock` in the repository, Cargo.lock speed up the dependencies trie resolution, and more importantly guarantees that everyone is using the same dependencies versions, without cargo.lock, developers may experience inconsistencies.

Reply:
So the ide specific and macos specific ignores I don't like. You should configure those in your global got settings, otherwise every developer will want to add their custom config

Reply:
When it comes to Cargo.lock I don't care much. I don't remember consciously ignoring it, but if I'd have to guess it was probably because rebase/merges are annoying and the added/removed lines of code in a PR are scewed

Reply:
> You should configure those in your global got settings, otherwise every developer will want to add their custom config

I noticed that [Rust gitignore](https://github.com/rust-lang/rust/blob/master/.gitignore#L13-L24), [Substrate gitignore](https://github.com/paritytech/substrate/blob/polkadot-v0.9.43/.gitignore#L12C1-L15) and [Solana Gitignore](https://github.com/solana-labs/solana/blob/v1.16.2/.gitignore#L22-L25) includes Editor and System specific stuff, but I'm ok on remove those.

Reply:
> When it comes to Cargo.lock I don't care much

Perfect, in this case I'll uncomment it and commit the Cargo.lock, I had issues in the past where the `Cargo.toml` was tracking a branch from Substrate repository, then suddently it stop working, it was because they `Cargo.lock` was created after breaking changes being introduced in the branch, devs who had the old `Cargo.lock` the project was working fine.